bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTable.java

```diff
@@ -316,10 +355,48 @@ public CompletableFuture<Result> increment(Increment increment) {
         clientWrapper.readModifyWriteRowAsync(hbaseAdapter.adapt(increment)));
   }
 
+  // NOTE: At HBase 2.4, the return type changed from CompletableFuture<Void> to
+  // CompletableFuture<Result>. The behavior also changed: the parameter can now accept
+  // Increments and Appends and will return the result of the Increment/Appends. When increment
+  // or Append is not present, the return value is now Result.EMPTY
   /** {@inheritDoc} */
   @Override
-  public CompletableFuture<Void> mutateRow(RowMutations rowMutations) {
-    return toCompletableFuture(clientWrapper.mutateRowAsync(hbaseAdapter.adapt(rowMutations)));
+  public CompletableFuture /*<Void|Result>*/ mutateRow(RowMutations rowMutations) {
+    Object emptyReturn = MUTATE_ROW_RETURNS_RESULT ? Result.EMPTY_RESULT : null;
+
+    if (rowMutations.getMutations().isEmpty()) {
+      return CompletableFuture.completedFuture(emptyReturn);
```

Comment:
should we throw exception in this case? Or this is acceptable in hbase?

Reply:
no, this is the expected behavior